cdap-ui/app/directives/metric-picker/metric-picker.js

```diff
@@ -109,45 +145,54 @@ angular.module(PKG.name + '.commons')
         elem.find('button').on('blur', onBlurHandler);
         elem.find('input').on('blur', onBlurHandler);
 
-        scope.$watchCollection('metric', function (newVal, oldVal) {
+        var metricChanged = function (newVal, oldVal) {
           ngModel.$validate();
 
           if(newVal.type !== oldVal.type) {
             scope.metric.context = '';
-            scope.metric.name = '';
+            scope.metric.resetNames();
```

Comment:
Right now, it'll remove the 4 rows, and clear the first.
Should we keep the metrics and not reset them?